test/powershell/Modules/Microsoft.PowerShell.Utility/Sort-Object.Tests.ps1

```diff
@@ -21,204 +21,946 @@
+        $employee2 = [pscustomobject]@{"FirstName"="joesph"; "LastName"="smith"; "YearsInMS"=15}
+        $employee3 = [pscustomobject]@{"FirstName"="john"; "LastName"="smyth"; "YearsInMS"=12}
+        $employees = @($employee1,$employee2,$employee3)
+        $results = $employees | Sort-Object -Property "LastName","FirstName" -Descending:$false
+        
+        $results[0] | Should Be $employees[1]
+        $results[1] | Should Be $employees[0]
+        $results[2] | Should Be $employees[2]
+    }
+
+    It "Sort-Object with -Descending and Two Order Entry Keys should work"{
+        $employee1 = [pscustomobject]@{"FirstName"="john"; "LastName"="smith"; "YearsInMS"=5}
+        $employee2 = [pscustomobject]@{"FirstName"="joesph"; "LastName"="smith"; "YearsInMS"=15}
+        $employee3 = [pscustomobject]@{"FirstName"="john"; "LastName"="smyth"; "YearsInMS"=12}
+        $employees = @($employee1,$employee2,$employee3)
+        $results = $employees | Sort-Object -Property "LastName","FirstName" -Descending
+        
+        $results[0] | Should Be $employees[2]
+        $results[1] | Should Be $employees[0]
+        $results[2] | Should Be $employees[1]
+    }
+
+    It "Sort-Object with Two Order Entry Keys with asc=true should work"{
+        $employee1 = [pscustomobject]@{"FirstName"="john"; "LastName"="smith"; "YearsInMS"=5}
+        $employee2 = [pscustomobject]@{"FirstName"="joesph"; "LastName"="smith"; "YearsInMS"=15}
+        $employee3 = [pscustomobject]@{"FirstName"="john"; "LastName"="smyth"; "YearsInMS"=12}
+        $employees = @($employee1,$employee2,$employee3)
+        $ht1 = @{"e"="FirstName"; "asc"=$true}
+        $ht2 = @{"e"="LastName";}
+        $results = $employees | Sort-Object -Property @($ht1,$ht2) -Descending
+        
+        $results[0] | Should Be $employees[1]
+        $results[1] | Should Be $employees[2]
+        $results[2] | Should Be $employees[0]
+    }
+
+    It "Sort-Object with Descending No Property should work"{
+        $employee1 = 1
+        $employee2 = 2
+        $employee3 = 3
+        $employees = @($employee1,$employee2,$employee3)
+        $results = $employees | Sort-Object -Descending
+        
+        $results[0] | Should Be 3
+        $results[1] | Should Be 2
+        $results[2] | Should Be 1
+    }
 }
+
+Describe 'Sort-Object Top and Bottom Unit Tests' -Tags 'CI' {
+
+    It 'Return the top N integers sorted in ascending order' {
+        $unsortedData = 973474993,271612178,-1258909473,659770354,1829227828,-1709391247,-10835210,-1477737798,1125017828,813732193
+        $baseSortParameters = @{}
+        $fullSortResults = $unsortedData | Sort-Object @baseSortParameters
+        $topNSortResults = $unsortedData | Sort-Object @baseSortParameters -Top 3
+        for ($i = 0; $i -lt $topNSortResults.Count; $i++) {
+            $topNSortResults[$i] | Should Be $fullSortResults[$i]
+        }
+    }
```

Comment:
I had written a function at first, with the tests being simply one or two lines of code; this resulted in a complicated function which would have to be tested by itself. I really didn't like this model, and preferred a short test with a half-dozen lines of code or so, which allows anyone to review a failing test in place and see more easily why it may be failing, plus enable testing outside of Pester by simply copying and pasting the test contents.